graylog2-server/src/main/java/org/graylog/scheduler/JobSchedulerService.java

```diff
@@ -61,8 +90,13 @@ protected void run() throws Exception {
                 LOG.debug("Starting scheduler loop iteration");
                 try {
                     if (!jobExecutionEngine.execute() && isRunning()) {
-                        // We wait because there are either no free worker threads or no runnable triggers
-                        clock.sleepUninterruptibly(1, TimeUnit.SECONDS); // TODO: Duration should be configurable
+                        // When the execution engine returned false, there are either no free worker threads or no
+                        // runnable triggers. To avoid busy spinning we sleep for the configured duration or until
+                        // we receive a job completion event via the scheduler event bus.
+                        if (sleeper.sleep(loopSleepDuration.getQuantity(), loopSleepDuration.getUnit())) {
```

Comment:
This can throw `InterruptedException` which will cause an error to be logged below.
Since interruptions can also happen spuriously, those aren't actually an error.
Can we add another catch arm here and log on trace or debug for those?
All other Exceptions are fine (even if none should be thrown here) on LOG.error, afaics.